transport/http/config.go

```diff
@@ -81,21 +81,50 @@ func (ts *transportSpec) Spec() config.TransportSpec {
 // 	transports:
 // 	  http:
 // 	    keepAlive: 30s
+//      maxIdleConnsPerHost: 2
+//      connTimeout: 500ms
+//      connBackoff:
+//        exponential:
+//          min: 100ms
+//          base: 100ms
+//          max: 30s
 //
 // All parameters of TransportConfig are optional. This section may be omitted
 // in the transports section.
 type TransportConfig struct {
 	// Specifies the keep-alive period for all HTTP clients. This field is
 	// optional.
-	KeepAlive time.Duration `config:"keepAlive"`
+	KeepAlive           time.Duration  `config:"keepAlive"`
+	MaxIdleConnsPerHost int            `config:"maxIdleConnsPerHost"`
+	ConnTimeout         time.Duration  `config:"connTimeout"`
+	ConnBackoff         config.Backoff `config:"connBackoff"`
 }
 
 func (ts *transportSpec) buildTransport(tc *TransportConfig, k *config.Kit) (transport.Transport, error) {
+	options := newTransportOptions()
+
 	opts := ts.TransportOptions
+	for _, opt := range opts {
+		opt(&options)
```

Comment:
nit: don't use a local variable for `ts.TransportOptions`, especially ith the name `opts`. High chance of confusing it with `options`.